### DISCUSSION TITLE
Deploying a contract via web3modal and wagmi.sh?

### DISCUSSION THREAD

**Discussion by malachy17** [WalletConnect/walletconnect-monorepo#3697] [Developer Support] [answered]
How would I go about deploying a contract via Web3Modal and Wagmi.sh? 

I don't see any functions within wagmi core that allows me to deploy a contract. Is there an easy way to do this that I'm not considering? Would I have to use another library such as Viem or Ethers.js?

Thank you.

**Comment by glitch-txs** [comment, marked as answer]
You can use Viem as well as Wagmi with Web3Modal, both libraries work hand in hand

**Reply by malachy17** [reply]
Thank you!
